blog/content/post/2025-07-29-zarthusbmc-se.md

```diff
@@ -0,0 +1,455 @@
+
+The GPIO definitions for `x11ssh` for
+[u-bmc](https://github.com/osresearch/u-bmc/blob/kf/x11/platform/supermicro-x11ssh-f/pkg/gpio/platform.go)[^keno-ubmc],
+[HardenedVault's attempt](https://github.com/hardenedvault/openbmc/blob/x11ssh-f/meta-supermicro/meta-x11ssh/recipes-kernel/linux/linux-aspeed/0001-add-aspeed-bmc-supermicro-x11ssh-dts.patch)[^hw-attempt],
+or our attempts, all use the work of [Keno Fisher](https://github.com/keno)[^keno]
+for those definitions. While the subset for pins might have been sufficient for
+`u-bmc` in the past (the x11ssh development has been abandoned), it is not
+sufficient for now. I've got to admit, it's still a mystery to me how Keno
+managed to come up with those definitions; however, I'll touch on one possible
+approach later in this post.
+
+## The effort
+
+The "damage control" isn't the only thing we were focusing on since the last
+time. Although we have not yet been successful in addressing the previously
+mentioned issues, as many other projects come in the way, we still managed to
+progress a lot thanks to the community effort. Let's discuss what we managed to
+achieve.
+
+### Open discussions
+
+Let's start with something different, non-technical. We've launched
+[ZarhusBMC discussions](https://github.com/zarhus/zarhusbmc/discussions)[^dscsns]
+where anyone interested in the project can check out what we are currently
+working on in the BMC space. In true open-source spirit, we want to be as
+transparent as possible, and we encourage taking an active part in the
+discussion.
+
+We've already hosted two such discussions. These aim for these to be a more
+streamlined form of live coding. Every time we're working on something, a
+discussion is made, and we update statuses as we go. This has two upsides that
+are partially related. First of all, we're closer to the community, and
+secondly, in case we require any 3rd party or community input, the results and
+steps we tried are in the commonplace and publicly available.
+
+Speaking of commonplace, we've chosen GitHub discussions as a place to host
+these. The reason is, once again, openness. There's no need to have any accounts
+to get updates on the topics, and since this is a very developer-oriented
+topic, most of the folks interested in this kind of stuff already have a GitHub
+account to take part in the discussion.
+
+As a side note, being so open also has its downsides. One being the free
+(as in free will) content is a
+"[free real estate](https://i.imgflip.com/24r48o.jpg?a486960)[^meme]" for AI
+bot farms, which we
+[already had a taste of](https://youtu.be/kK6Dz1gnmmY)[^bots] 😉.
+
+### Probing stock firmware p. I - QEMU, binwalk, gdb, and stock firmware sources
+
+In order to resolve the previously mentioned issues, mainly the GPIO pins
+definitions, we decided to give a stock firmware probing a chance. The attempt
+can be summarized in 4 steps:
+
+1. First, we decided to have a go at disassembling the firmware binary. The
```

Comment:
Instead of numbering, I would add an h4 section, unless it doesn't render well. Just a suggestion.